### DISCUSSION TITLE
Type alias without all type arguments specified (type Alias MyList = List)

### DISCUSSION THREAD

**Discussion by edwardpeters** [finos/morphir#196] [Q&A]
If I try to compile
`type alias MyList = List`
in native elm, it fails with
`The `List` type needs 1 argument, but I see 0 instead:`
The same in morphir-elm compiles without issue, but I have not found any way to create a valid example (I cannot seem to instantiate a `MyList Int`, for instance

```
type alias MyList  = List 

foo : (MyList Int) -> Int
foo x = 1

bar : () -> Int
bar () = foo [1]
```

I'm not sure whether this functionality - allowing aliasing of generic types without all parameters specified - is intended or not. Is this a case where the compiler lets invalid type definitions through, or is there a valid implementation/use of the type as defined that I just haven't found?

On a general note, is there a place where semantics and type rules are documented at this level of specificity?
(For context, I'm working on a scala type checker and need to know if I should support this case.)

**Comment by AttilaMihaly** [comment]
It's not intended. This is a missing check in the Elm frontend. Originally we started out by making the assumption that users call the Elm compiler before running `morphir make` to ensure all the logic is valid. We gradually added a lot of checks over time but we didn't invest into full coverage yet.

We don't have semantics documented to this level. Unless stated otherwise Elm's semantics apply for now.

**Reply by edwardpeters** [reply]
Thanks! A few follow ups:

1. I have frequently been unable to call `morphir make` due to the presence of imports from the Morphir SDK - is there a convenient way to do that?
2. Does "Unless otherwise stated" refer specifically to the documentation at https://package.elm-lang.org/packages/finos/morphir-elm/20.1.0/ , or is there another place I should be looking?
3. When I find things in this category (missing checks), is it worth filing a bug report with morphir-elm?
